Further remove MSVC code

Constellation opened the pull request:
#### adcd97e5a4ae359e0ea83a4cc9ae2348b95d607f
<pre>
Further remove MSVC code
<a href="https://bugs.webkit.org/show_bug.cgi?id=274450">https://bugs.webkit.org/show_bug.cgi?id=274450</a>
<a href="https://rdar.apple.com/128448731">rdar://128448731</a>

Reviewed by Ross Kirsling.

Further remove MSVC hacks. And we use `OS(WINDOWS)` for code which needs to be used on Windows (even without MSVC).

* Source/JavaScriptCore/API/tests/testapi.c:
* Source/JavaScriptCore/API/tests/testapi.cpp:
* Source/JavaScriptCore/assembler/MacroAssemblerX86Common.cpp:
(JSC::MacroAssembler::probe):
(JSC::MacroAssemblerX86Common::getCPUIDEx):
* Source/JavaScriptCore/assembler/X86Assembler.h:
* Source/JavaScriptCore/builtins/BuiltinNames.cpp:
* Source/JavaScriptCore/config.h:
* Source/JavaScriptCore/jit/PCToCodeOriginMap.cpp:
* Source/JavaScriptCore/jit/RegisterAtOffset.cpp:
* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
(jscmain):
* Source/JavaScriptCore/offlineasm/generate_offset_extractor.rb:
* Source/JavaScriptCore/parser/SourceProviderCacheItem.h:
* Source/JavaScriptCore/runtime/JSCPtrTag.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/JavaScriptCore/testRegExp.cpp:
* Source/WTF/wtf/Assertions.cpp:
(WTF::createWithFormatAndArguments):
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/MathExtras.h:
* Source/WTF/wtf/PtrTag.h:
* Source/WTF/wtf/SegmentedVector.h:
* Source/WTF/wtf/posix/ThreadingPOSIX.cpp:

Canonical link: <a href="https://commits.webkit.org/279108@main">https://commits.webkit.org/279108@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c8b90f94fbe592b798e3471e962ff55eee8a9ff0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52510 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31844 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4934 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55784 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3233 "Built successfully") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/38421 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2933 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42671 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2062 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54607 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29493 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45314 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23760 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26671 "Unexpected infrastructure issue, retrying build") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2590 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1392 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Compiled WebKit") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/45859 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48562 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2744 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57380 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/52020 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27642 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2765 "Found 3 new test failures: imported/w3c/web-platform-tests/css/css-view-transitions/animating-new-content-subset.html, imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-and-on-top-of-viewport-partially-onscreen-old.html, imported/w3c/web-platform-tests/css/css-view-transitions/transform-origin-view-transition-group.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50067 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28870 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45432 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49322 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29782 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/64326 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7702 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28619 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12182 "Found 10 new JSC stress test failures: stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.bytecode-cache, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.default, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.dfg-eager, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.dfg-eager-no-cjit-validate, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.eager-jettison-no-cjit, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.lockdown, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.mini-mode, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.no-cjit-collect-continuously, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.no-cjit-validate-phases, stress/dfg-ai-direct-get-by-id-attribute-change-transition.js.no-llint (failure)") | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Source/WTF/wtf/posix/ThreadingPOSIX.cpp:
```diff
@@ -53,12 +53,6 @@
 #endif
 #endif
 
-#if !COMPILER(MSVC)
-#include <limits.h>
-#include <sched.h>
-#include <sys/time.h>
-#endif
```
Hmm, this is a `!` but I take it we don't need to keep them? (I guess `<sched.h>` is already included under `OS(LINUX)`...)

Constellation replied:
This is ThreadingPOSIX.cpp, so it is not used in Windows.

rkirsling replied:
Of course; I'm saying the `#if` has a `!`, so one would expect that we need to keep all of these includes.

rkirsling reviewed (approved):
r=me with a comment

Constellation committed:
Further remove MSVC code
https://bugs.webkit.org/show_bug.cgi?id=274450
rdar://128448731

Reviewed by Ross Kirsling.

Further remove MSVC hacks. And we use `OS(WINDOWS)` for code which needs to be used on Windows (even without MSVC).

* Source/JavaScriptCore/API/tests/testapi.c:
* Source/JavaScriptCore/API/tests/testapi.cpp:
* Source/JavaScriptCore/assembler/MacroAssemblerX86Common.cpp:
(JSC::MacroAssembler::probe):
(JSC::MacroAssemblerX86Common::getCPUIDEx):
* Source/JavaScriptCore/assembler/X86Assembler.h:
* Source/JavaScriptCore/builtins/BuiltinNames.cpp:
* Source/JavaScriptCore/config.h:
* Source/JavaScriptCore/jit/PCToCodeOriginMap.cpp:
* Source/JavaScriptCore/jit/RegisterAtOffset.cpp:
* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
(jscmain):
* Source/JavaScriptCore/offlineasm/generate_offset_extractor.rb:
* Source/JavaScriptCore/parser/SourceProviderCacheItem.h:
* Source/JavaScriptCore/runtime/JSCPtrTag.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/JavaScriptCore/testRegExp.cpp:
* Source/WTF/wtf/Assertions.cpp:
(WTF::createWithFormatAndArguments):
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/MathExtras.h:
* Source/WTF/wtf/PtrTag.h:
* Source/WTF/wtf/SegmentedVector.h:
* Source/WTF/wtf/posix/ThreadingPOSIX.cpp:

Canonical link: https://commits.webkit.org/279108@main
